src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsNuGetPathContextFactory.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnvDTE;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsNuGetPathContextFactory))]
+    public class VsNuGetPathContextFactory : IVsNuGetPathContextFactory
+    {
+        private readonly ISettings _settings;
+        private readonly IVsSolutionManager _solutionManager;
+
+        [ImportingConstructor]
+        public VsNuGetPathContextFactory(ISettings settings, IVsSolutionManager solutionManager)
```

Comment:
Sounds great. Didn't realize MEF supports this out of the box. So you can `Lazy<T>` import any type>